testnet/stacks-node/src/tests/signer/commands/block_commit.rs

```diff
@@ -0,0 +1,89 @@
+use super::context::{SignerTestContext, SignerTestState};
+use crate::tests::signer::v0::MultipleMinerTest;
+use madhouse::{Command, CommandWrapper};
+use proptest::prelude::{Just, Strategy};
+use std::sync::{Arc, Mutex};
+
+pub struct SubmitBlockCommitSecondaryMinerCommand {
+    miners: Arc<Mutex<MultipleMinerTest>>,
+}
+
+impl SubmitBlockCommitSecondaryMinerCommand {
+    pub fn new(miners: Arc<Mutex<MultipleMinerTest>>) -> Self {
+        Self { miners }
+    }
+}
+
+impl Command<SignerTestState, SignerTestContext> for SubmitBlockCommitSecondaryMinerCommand {
+    fn check(&self, state: &SignerTestState) -> bool {
+        println!(
+            "Checking: Submitting block commit miner 2. Result: {:?}",
+            state.is_secondary_miner_skip_commit_op
+        );
+        state.is_secondary_miner_skip_commit_op
+    }
+
+    fn apply(&self, _state: &mut SignerTestState) {
```

Comment:
Consider `exec` or `execute` instead of `apply` as it's not necessarily "applying" something to the `State`, e.g. in the case of `WaitForBlockFromMiner1Command` etc.

Reply:
@cylewitruk, I'd open an issue in madhouse-rs to consider it separately.